tests/Neo.Json.UnitTests/UT_JArray.cs

```diff
@@ -252,7 +252,7 @@ public void TestAsString()
                 bob,
             };
             var s = jArray.AsString();
-            Assert.AreEqual(s, "{\"name\":\"alice\",\"age\":30,\"score\":100.001,\"gender\":\"female\",\"isMarried\":true,\"pet\":{\"name\":\"Tom\",\"type\":\"cat\"}},{\"name\":\"bob\",\"age\":100000,\"score\":0.001,\"gender\":\"male\",\"isMarried\":false,\"pet\":{\"name\":\"Paul\",\"type\":\"dog\"}}");
+            Assert.AreEqual(s, "[{\"name\":\"alice\",\"age\":30,\"score\":100.001,\"gender\":\"female\",\"isMarried\":true,\"pet\":{\"name\":\"Tom\",\"type\":\"cat\"}},{\"name\":\"bob\",\"age\":100000,\"score\":0.001,\"gender\":\"male\",\"isMarried\":false,\"pet\":{\"name\":\"Paul\",\"type\":\"dog\"}}]");
```

Comment:
it change the behavior

Reply:
And JSON is available from contracts.

Reply:
If it must be a hardfork, then make it a hardfork, this one also need to be reviewed:

https://github.com/neo-project/neo/pull/3037

Reply:
change the contact behavior is bad,,,,,,,i know, but what we gonna do with tons of problems in the core....

Reply:
> And JSON is available from contracts.

BTW, how does contract use this?

Reply:
Stdlib's jsonSerialize/jsonDeserialize.

Reply:
> Stdlib's jsonSerialize/jsonDeserialize.

since this is a bug of c#, are you sure neogo will have the same behavior?

Reply:
My experience is that `As` and `To` for Neo.Json objects behave differently. So its correct that `AsString` does this, it gets the inner objects as `string` without `[]`. Where `ToString` would return the `[]` for the collection.

Reply:
no, except this one, they basically the same:

```C#
  public override string AsString()
        {
            return Value.ToString().ToLowerInvariant();
        }
```

```C#
       public override string AsString()
        {
            return Value.ToString(CultureInfo.InvariantCulture);
        }
```

```C#
  public virtual string AsString()
    {
        return ToString();
    }
```

Reply:
And without `[]` is only part of the problem, it also lacks `null` object.

Reply:
Than what is the point of having `AsString` vs `ToString`? If they both do the samething.

Reply:
> Than what is the point of having `AsString` vs `ToString`? If they both do the samething.

To be honest i dont know,,,, and there is no document for this

Reply:
> And without `[]` is only part of the problem, it also lacks `null` object.

could you fix both at same time, but we need to ensure that is not possible to call it by a smart contract

Reply:
> > And without `[]` is only part of the problem, it also lacks `null` object.
> 
> could you fix both at same time, but we need to ensure that is not possible to call it by a smart contract

Already fixed.